tst/regression/utils/test_case.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
+#========================================================================================
+# Athena++ astrophysical MHD code
+# Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+# Licensed under the 3-clause BSD License, see LICENSE file for details
+#========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#========================================================================================
+
+import os
+from shutil import rmtree
+import subprocess
+
+class Parameters():
+    def __init__(self):
+        driver_path = ""
+        driver_input_path = ""
+        test_path = ""
+        output_path = ""
+        mpi_cmd = ""
+        mpi_opts = ""
+
+class TestCase:
```

Comment:
A model I like better than defining `analyze` as a function in each test script is to make `TestCase` an abstract base class, with `analyze` as an undefined member function. Then every test defines a new class which inherits from `TestCase` and defines `analyze`.

Reply:
That's probably a better design.